src/tool/sandbox.py

```
@@ -51,3 +61,38 @@ def write_file(filepath: str, content: str) -> str:
         return f"Error writing to file: {str(e)}"
+
+@tool
+def execute_python_file(filepath: str, timeout_seconds: int = 5) -> str:
+    """执行指定的 Python 文件并返回 stdout/stderr (Execute a Python file and capture stdout/stderr)."""
+    try:
+        if not os.path.exists(filepath):
+            return f"Error: File {filepath} does not exist."
+        if not filepath.endswith(".py"):
+            return f"Error: File {filepath} is not a Python file."
+
+        completed = subprocess.run(
```

Ruff (0.15.2)

[error] `subprocess` call: check for execution of untrusted input

(S603)